app/models/terminal_executor.rb

```diff
@@ -34,26 +33,26 @@ def stop!(signal)
   private
 
   def execute_command!(command)
-    output, input, @pid = Bundler.with_clean_env do
-      PTY.spawn(command, in: '/dev/null')
-    end
+    Open3.popen2e(whitelisted_env, command, in: '/dev/null', unsetenv_others: true, pgroup: true)  do |stdin, oe, wait_thr|
+      @pid = wait_thr.pid
 
-    @pgid = begin
-      Process.getpgid(pid)
-    rescue Errno::ESRCH
-      nil
-    end
+      @pgid = begin
+        Process.getpgid(@pid)
+      rescue Errno::ESRCH
+        nil
+      end
 
-    begin
-      output.each(256) {|line| @output.write(line) }
-    rescue Errno::EIO
-      # The IO has been closed.
-    end
+      oe.each(256) do |line|
+        @output.write line.gsub("\n", "\r\n")
+      end
 
-    _, status = Process.wait2(pid)
-
-    input.close
+      wait_thr.value
+    end.success?
+  end
 
-    return status.success?
+  def whitelisted_env
+    whitelist = ['TERM', 'SHELL'] + ENV.keys.select { |k| k.end_with?('PATH', 'DIR', '_HOME') }
```

Comment:
Seems kind of arbitrary that we're keeping env vars that end with `PATH`, `DIR`, or `_HOME`.

I would rather have this whitelist be minimal and documented, with only the ENV vars that would be necessary for the commands to function.